### PR TITLE
BIP15x refactor

### PR DESCRIPTION
**Pull request by sergey-chernikov**
BIP15x code can now be used in any connection - specifically it was required to replace SSL with BIP15x in WebSockets connection. It should work with ZMQ as well (which will be obsoleted soon).

### CODE REVIEW

File: BlocksettleNetworkingLib/BIP15xHelpers.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*
+
+***********************************************************************************
+* Copyright (C) 2018 - 2020, BlockSettle AB
```
Comment (Ation) [inline BlocksettleNetworkingLib/BIP15xHelpers.h:4]: It's a new file. Please put 2020 - 2020

Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/BIP15xHelpers.h:4]: No, actually it's a renamed ZMQ_BIP15X_Helpers.h, which was originally created in 2018, I suppose.

File: BlocksettleNetworkingLib/BIP15xMessage.h [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+         public:
+            // Parse and return immutable message.
+            // Message might be invalid if parsing failed (check with isValid)
+            // Does not copy underlying raw data, make sure packet is live long enough.
+            static Message parse(const BinaryDataRef& packet);
+
+            // Validate if packet is valid before use
+            bool isValid() const { return type_ != MsgType::Undefined; }
+
+            // Packet's type (ZMQ_MSGTYPE_SINGLEPACKET, ZMQ_MSGTYPE_HEARTBEAT etc)
```
Comment (goatpig) [inline BlocksettleNetworkingLib/BIP15xMessage.h:101]: left over zmq nomenclature in comment

File: BlocksettleNetworkingLib/ChatProtocol/ChatClientLogic.cpp [hunk 3]
```diff
@@ -14,6 +14,8 @@
 #include "ChatProtocol/ChatClientLogic.h"
 
 #include "ConnectionManager.h"
+#include "DataConnection.h"
+#include "ZmqDataConnection.h"
```
Comment (goatpig) [inline BlocksettleNetworkingLib/ChatProtocol/ChatClientLogic.cpp:18]: chat still uses zmq?

Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/ChatProtocol/ChatClientLogic.cpp:18]: Yes, though switching to WsDataConnection can be performed in a couple of line change (in another PR, I think).

File: BlocksettleNetworkingLib/BIP15xHelpers.h (hunk 1 above)
Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/BIP15xHelpers.h:4]: Fixed year in the only new file (Transport.h)

**Pull request merged**